pkg/aws/aws_client/role.go

```diff
@@ -89,3 +96,202 @@ func (client *AWSClient) ListRoles() ([]types.Role, error) {
 	out, err := client.IamClient.ListRoles(context.TODO(), input)
 	return out.Roles, err
 }
+
+func (client *AWSClient) PolicyAttachedToRole(roleName string, policyArn string) (bool, error) {
```

Comment:
NP update the func name to reflect what we are checking
```suggestion
func (client *AWSClient) IsPolicyAttachedToRole(roleName string, policyArn string) (bool, error) {
```